### PR TITLE
EGamma: add scale & systematics variables to electron and photon table backport

### PR DESCRIPTION
**Pull request by sroychow**
#### PR description:
This is a backport of #33030 *only the part related to NanoAOD. 

#### PR validation:
Checked with wf - 136.8523,136.8522

@mariadalfonso @jainshilpi

### CODE REVIEW

File: PhysicsTools/NanoAOD/python/photons_cff.py [hunk 1]
```diff
@@ -73,36 +73,39 @@ def make_bitmapVID_docstring(id_modules_working_points_pset):
...
-    produceCalibratedObjs = False,
+run2_nanoAOD_94XMiniAODv2.toModify(calibratedPatPhotonsNano,
     correctionFile = cms.string("EgammaAnalysis/ElectronTools/data/ScalesSmearings/Run2017_17Nov2017_v1_ele_unc")
 )
 
-calibratedPatPhotons80XLegacy = RecoEgamma.EgammaTools.calibratedEgammas_cff.calibratedPatPhotons.clone(
-    produceCalibratedObjs = False,
-    correctionFile = cms.string("EgammaAnalysis/ElectronTools/data/ScalesSmearings/Legacy2016_07Aug2017_FineEtaR9_v3_ele_unc"),
-)
-
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/python/photons_cff.py:105]: we added this in master ```
run2_miniAOD_80XLegacy.toModify(calibratedPatPhotonsNano,
                                 correctionFile = cms.string("EgammaAnalysis/ElectronTools/data/ScalesSmearings/Legacy2016_07Aug2017_FineEtaR9_v3_ele_unc")
 )
```

File: PhysicsTools/NanoAOD/python/photons_cff.py [hunk 2]
```diff
@@ -313,11 +326,11 @@ def make_bitmapVID_docstring(id_modules_working_points_pset):
 from RecoEgamma.PhotonIdentification.photonIDValueMapProducer_cff import photonIDValueMapProducer
 ###UL to be done first
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/python/photons_cff.py:327]: we had this in master
```
 _withUL16preVFPScale_sequence = photonSequence.copy()
 _withUL16preVFPScale_sequence.replace(slimmedPhotonsWithUserData, calibratedPatPhotonsNano  + slimmedPhotonsWithUserData)
 (run2_egamma_2016 & tracker_apv_vfp30_2016).toReplaceWith(photonSequence, _withUL16preVFPScale_sequence)

 _withUL16postVFPScale_sequence = photonSequence.copy()
 _withUL16postVFPScale_sequence.replace(slimmedPhotonsWithUserData, calibratedPatPhotonsNano  + slimmedPhotonsWithUserData)
 (run2_egamma_2016 & ~tracker_apv_vfp30_2016).toReplaceWith(photonSequence, _withUL16postVFPScale_sequence)

```

**Comment by mariadalfonso** [comment]
plese test

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
please test

File: PhysicsTools/NanoAOD/python/electrons_cff.py [hunk 3]
```diff
@@ -248,56 +243,22 @@ def _get_bitmapVIDForEle_docstring(modules,WorkingPoints):
...
-run2_egamma_2018.toModify(slimmedElectronsWithUserData.userFloats,
-    ecalTrkEnergyErrPostCorrNew = cms.InputTag("calibratedPatElectronsUL18","ecalTrkEnergyErrPostCorr"),
-    ecalTrkEnergyPreCorrNew     = cms.InputTag("calibratedPatElectronsUL18","ecalTrkEnergyPreCorr"),
-    ecalTrkEnergyPostCorrNew    = cms.InputTag("calibratedPatElectronsUL18","ecalTrkEnergyPostCorr"),
-)
-
-run2_miniAOD_80XLegacy.toModify(slimmedElectronsWithUserData.userFloats,
-    mvaSpring16GP = cms.InputTag("electronMVAValueMapProducer:ElectronMVAEstimatorRun2Spring16GeneralPurposeV1Values"),
-    mvaSpring16HZZ = cms.InputTag("electronMVAValueMapProducer:ElectronMVAEstimatorRun2Spring16HZZV1Values"),
-)
```
Comment (mariadalfonso) [inline PhysicsTools/NanoAOD/python/electrons_cff.py:269]: we removed too much here.
 see master 
https://github.com/cms-sw/cmssw/pull/33030/files#diff-55e7cbee5e9d43b8ddb2363bf2c90ed2a0bdd09be6f8604d6b45fa088ec53856R283-R286

Comment (sroychow) [reply PhysicsTools/NanoAOD/python/electrons_cff.py]: Updated the config. Local test of wf 1329.1 passes.

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
for the (run2_nanoAOD_106Xv1 & ~run2_nanoAOD_devel) we still see the correction being applied for the 2016. Both photons and electrons are affected.

Here you undo correctly the addition of the scale variation, need to be extended also to remove the scale and systematic unc
https://github.com/cms-sw/cmssw/pull/33376/files#diff-55d52561f967f28c84e193cb47c37df21ed1f18e27ad2eb591ed9b5d06bce94cR308-L314

**Comment by mariadalfonso** [comment]
gentle ping @sroychow

**Comment by sroychow** [comment]
@mariadalfonso @jainshilpi I rebased the commit to an updated nightly. Can you check the last commit? This should contain the fix we discussed yesterday.

**Comment by mariadalfonso** [comment]
please test

**Comment by qliphy** [comment]
please test

**Comment by mariadalfonso** [comment]
@sroychow 

we miss the backport of the DQM
https://github.com/cms-sw/cmssw/pull/33030/files#diff-f767099a77b90dde5ea9b5f3805a3a255c0c7d1e050245f91f959efce87924d2

**Comment by mariadalfonso** [comment]
please test

**Comment by mariadalfonso** [comment]
backport

**Comment by mariadalfonso** [comment]
+xpog
backport inline with master
nanov8 unchanged

**Comment by qliphy** [comment]
+1

**Pull request merged**